core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/NetworkCommunicationActor.scala

```diff
@@ -65,15 +65,15 @@ class NetworkCommunicationActor(parentRef: ActorRef, val actorId: ActorVirtualId
   val queriedActorVirtualIdentities = new mutable.HashSet[ActorVirtualIdentity]()
   val messageStash = new mutable.HashMap[ActorVirtualIdentity, mutable.Queue[WorkflowMessage]]
   val messageIDToIdentity = new mutable.LongMap[ActorVirtualIdentity]
-  //register timer for resending messages
-  val resendHandle: Cancellable = context.system.scheduler.schedule(
+  // register timer for resending messages
+  val resendHandle: Cancellable = context.system.scheduler.scheduleWithFixedDelay(
```

Comment:
I just checked Akka's doc and `scheduleWithFixedDelay` is indeed the recommended way

Reply:
sounds good